### DISCUSSION TITLE
Codegen missing import when having an array of objects

### DISCUSSION THREAD

**Discussion by alfonsocj** [reactwg/react-native-new-architecture#104] [Q&A]
Tested `react-native-codegen` version 0.70.5

When having an array of objects prop like this one:
```
type ColorFilterStruct = Readonly<{
  keypath: string;
  color: ProcessedColorValue;
}>;

export interface NativeProps extends ViewProps {
  colorFilters?: ReadonlyArray<ColorFilterStruct>;
}
```

the generated `Props.h` misses the include `<react/renderer/core/propsConversions.h>` needed for structs, producing an error like: ![image](https://user-images.githubusercontent.com/3791120/208957604-1247af73-b31d-4c60-a767-ddbc69faaab8.png)

So far, I see that the problem is that codegen only adds `propsConversions.h` if there's a prop with `typeAnnotation.type === 'ObjectTypeAnnotation'` and not for `typeAnnotation.type === 'ArrayTypeAnnotation' && typeAnnotation.elementType.type === 'ObjectTypeAnnotation'`.

Adding a dummy object prop like the one below unblocks the issue since `Props.h` will include `propsConversion.h`.

```
export interface NativeProps extends ViewProps {
  colorFilters?: ReadonlyArray<ColorFilterStruct>;
  dummy?: Readonly<{ dummy: boolean }>;
}
```

@cipolleschi, I'm creating this issue here to increase visibility since others might have a similar problem.

**Comment by cipolleschi** [comment]
Thanks for doing this! I really hope to find some time to fix this. Luckily, being a patch, we may be able to release a fix for it in 0.71 without a release of React Native, thanks to the `^` notation in the dependencies.
